python/tvm/relay/op/nn/_nn.py

```diff
@@ -141,20 +142,29 @@ def convert_conv2d(attrs, inputs, tinfos, desired_layout):
     from tvm import relay
     data, weight = inputs
     new_attrs = dict(attrs)
-    new_attrs['data_layout'] = desired_layout
-    if desired_layout == 'NCHW':
+    desired_data_layout = str(desired_layouts[0])
+    assert desired_data_layout != "default", "Data layout cannot be default"
+    new_attrs['data_layout'] = desired_data_layout
+
+    if len(desired_layouts) > 1:
+        desired_kernel_layout = str(desired_layouts[1])
+        if desired_kernel_layout != "default":
+            new_attrs['kernel_layout'] = desired_kernel_layout
+            return relay.nn.conv2d(data, weight, **new_attrs)
+
+    # Handle default kernel layouts
+    if desired_data_layout == 'NCHW':
         new_attrs['kernel_layout'] = 'OIHW'
         return relay.nn.conv2d(data, weight, **new_attrs)
-    elif desired_layout == 'NHWC':
+    elif desired_data_layout == 'NHWC':
         # Check for depthwise convolution.
         if is_depthwise_conv2d(data.shape, attrs['data_layout'], weight.shape,
                                attrs['kernel_layout'], attrs['groups']):
             new_attrs['kernel_layout'] = 'HWOI'
         else:
             new_attrs['kernel_layout'] = 'HWIO'
         return relay.nn.conv2d(data, weight, **new_attrs)
-    else:
-        assert "Layout %s is not yet supported." % (desired_layout)
+    assert "Layout %s is not yet supported." % (desired_data_layout)
     return None
```

Comment:
ditto.

Reply:
Ditto

Reply:
Apologies I forgot to mention, although the return is unreachable I added it to silence the linter. In this case, I think its needed, or we disable the check for this function?

Reply:
We should not disable the checker anyways. In this case you have two options.

1. Move the assertion out of the branch. This solution makes the last block on the top-level to avoid inconsistent return statement, but I visually do not like, so I prefer the second one.

```python
if desired_data_layout == 'NCHW':
  # do something
  return
assert desired_data_layout == 'NHWC'
# do something
return
```

2. Throw exception instead of assertion. In this case, linter can know everything after `raise` is unreachable and it won't complain.

```python
if desired_data_layout == 'NCHW':
  # do something
  return
if desired_data_layout == 'NHWC': # PEP8 standard suggests not using else when previous block has a return statement. Since TVM linter doesn't enforce this point, I'm fine with both.
  # do something
  return
raise RuntimeError(...)
```